pymodm/fields.py

```diff
@@ -1202,11 +1234,24 @@ def to_mongo(self, value):
     def __get__(self, inst, owner):
         MongoModelBase = _import('pymodm.base.models.MongoModelBase')
         if inst is not None and isinstance(inst, MongoModelBase):
-            raw_value = inst._data.get(self.attname, self.default)
-            if self.is_blank(raw_value):
-                return raw_value
-            python = self.to_python(raw_value)
-            # Cache retrieved value.
-            self.__set__(inst, python)
-            return python
+            try:
+                value = inst._data.get_python_value(
+                    self.attname, self.to_python)
+            except KeyError:
+                value = self.default
+
+            # Default/blank values can be returned immediately.
+            if self.is_blank(value):
+                return value
+
+            # If dereferencing externally.
+            if isinstance(value, dict):
+                # Try to convert the value into our document type.
+                try:
+                    return self.related_model.from_document(value)
+                except (ValueError, TypeError):
+                    pass
+
+            # Else.
+            return self.dereference_if_needed(value)
```

Comment:
Maybe we can just replace these lines with a call to `to_python()` since this is doing the same work?